inference-engine/include/details/ie_so_pointer.hpp

```diff
@@ -145,6 +145,10 @@ IE_SUPPRESS_DEPRECATED_END
         return *this;
     }
 
+    operator std::shared_ptr<Loader>&() noexcept {
+        return _so_loader;
+    }
```

Comment:
why do we need to return by reference?

Reply:
To extract the loader from plugin_ptr implicitly when we construct the network. Seems `const` is needed here